specs/altair/beacon-chain.md

```diff
@@ -212,7 +211,7 @@ class SyncAggregate(Container):
 ```python
 class SyncCommittee(Container):
     pubkeys: Vector[BLSPubkey, SYNC_COMMITTEE_SIZE]
-    pubkey_aggregates: Vector[BLSPubkey, SYNC_COMMITTEE_SIZE // SYNC_PUBKEYS_PER_AGGREGATE]
+    aggregate_pubkey: BLSPubkey
```

Comment:
Is `aggregate_pubkey` field necessary? It seems that the validator / light client can compute it in runtime. If we remove `aggregate_pubkey`, we can replace `SyncCommittee` container with just `sync_committee_pubkeys: Vector[BLSPubkey, SYNC_COMMITTEE_SIZE]`.

Reply:
the light client could compute the aggregate at runtime but this enables an optimization where the light client can _subtract_ pubkeys from the set in `pubkeys` against the `aggregate_pubkey` with the idea being they can save a bunch of BLS computation in the expected case (i.e. fairly full sync committee participation)

for example, if we get 80% participation in a given sync committee signature, the light client can simply "unaggregate" 20% of the pubkeys, rather than have to aggregate 80% of them. this savings is particularly important for resource-constrained environments like on-chain bridges etc

Reply:
> on-chain bridges

Ah, that's fair. 😄

BTW, since `get_sync_committee_indices` might return duplicate indices, the aggregate pubkey may include duplicate pubkeys. Implementers have to detect the duplicate before subtracting the pubkeys. Maybe worth noting the potential duplication that at `get_sync_committee` docstring (like `get_sync_committee_indices` docstring).